src/main/java/org/broadinstitute/hellbender/tools/reference/CheckReferenceCompatibility.java

```diff
@@ -0,0 +1,284 @@
+package org.broadinstitute.hellbender.tools.reference;
+
+import htsjdk.samtools.SAMSequenceDictionary;
+import htsjdk.samtools.SAMSequenceRecord;
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.vcf.VCFHeader;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.argumentcollections.SequenceDictionaryValidationArgumentCollection;
+import org.broadinstitute.hellbender.engine.FeatureDataSource;
+import org.broadinstitute.hellbender.engine.GATKPath;
+import org.broadinstitute.hellbender.engine.GATKTool;
+import org.broadinstitute.hellbender.engine.ReferenceDataSource;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.SequenceDictionaryUtils;
+import org.broadinstitute.hellbender.utils.tsv.DataLine;
+import org.broadinstitute.hellbender.utils.tsv.TableColumnCollection;
+import org.broadinstitute.hellbender.utils.tsv.TableWriter;
+import picard.cmdline.programgroups.ReferenceProgramGroup;
+
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.nio.file.Path;
+import java.util.*;
+
+/**
+ * Check a BAM/VCF for compatibility against specified references.
+ *
+ * <p>This tool generates a table analyzing the compatibility of a sequence file against provided references. The tool works to compare
+ * BAM/CRAMs (specified using the -I argument) as well as VCFs (specified using the -V argument) against provided
+ * reference(s), specified using the -references-to-compare argument. The table can be directed to a file or standard
+ * output using provided command-line arguments.
+ *</p>
```

Comment:
add a sentance or two here explaining the differences in behavior if md5s are or aren't present.